### DISCUSSION TITLE
Provide a better way to try out a given documentation sample

### DISCUSSION THREAD

**Discussion by vmutafov** [eclipse/dirigible#1279] [Ideas]
Currently, if you want to try out some API sample from https://www.dirigible.io/api/, you need to open the Dirigible, create a project, create some JS file and set its content to be the sample you want to try. 

This is not the best experience. In many documentations for other frameworks/projects, there is a way to open a Stackblitz or something similar and check a given sample by just clicking a button.

We could provide something similar and add a button "Try it out!" on each sample we have and when the user clicks it, we could open up a Dirigible instance with the code of the sample already loaded.

From my point of view, this shouldn't be something hard to achieve but we should be able to open anonymous unique Dirigible sessions and have the ability to load up an initial Dirigible JS project with some sample provided by the request to open the IDE.

**Comment by ThuF** [comment]
That would be a great idea!

Partially such experience is already available with the **Run on Dirigible** button in the README.md of few of our GitHub samples:
- https://github.com/dirigiblelabs/demo-eclipsecon2018-ide
- https://github.com/dirigiblelabs/demo-eclipsecon2018-edm-complex
- https://github.com/dirigiblelabs/demo-eclipsecon2018-bpm

The button redirects to a central [Deploy Manager](https://github.com/dirigiblelabs/ide-deploy-manager) which can trigger a `Git Clone` action in a selected Eclipse Dirigible instance.

**Comment by vmutafov** [comment]
Really nice! I'll take a look what would be the easiest way to load up samples. Either migrating the samples to Git repositories or extending the Deploy Manager to load up a sample by providing it as some request parameter

**Comment by ThuF** [comment]
From supportability point of view, it would be better to have a single Git repository containing all samples as separate projects (folders). Also we would need to extend the Deploy Manager (and the Git perspective as well) to be able to import only selected project(s) after Git Clone.
